File: examples/android/MSDKSampleAppJava/app/src/main/java/msdk/java/sample/history/HistoryFragment.java
package msdk.java.sample.history;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import msdk.java.sample.databinding.HistoryFragmentBinding;

public class HistoryFragment extends Fragment {

    private HistoryFragmentBinding binding;
    private HistoryViewModel model;

    public static HistoryFragment newInstance() {
        return new HistoryFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = HistoryFragmentBinding.inflate(inflater, container, false);

        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getContext());
        binding.historyList.setLayoutManager(layoutManager);
        HistoryAdapter adapter = new HistoryAdapter();
        binding.historyList.setAdapter(adapter);

        model = new ViewModelProvider(requireActivity()).get(HistoryViewModel.class);
        model.getHistory().observe(getViewLifecycleOwner(), adapter::setData);
    }
}

File: examples/android/CMeSdkJava/app_kt/src/main/kotlin/me/connect/sdk/java/samplekt/history/HistoryFragment.kt
package me.connect.sdk.java.samplekt.history

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import me.connect.sdk.java.samplekt.databinding.HistoryFragmentBinding
import me.connect.sdk.java.samplekt.db.entity.Action

class HistoryFragment : Fragment() {
    private lateinit var binding: HistoryFragmentBinding
    private val model: HistoryViewModel by viewModels()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        binding = HistoryFragmentBinding.inflate(inflater, container, false)

        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val layoutManager: RecyclerView.LayoutManager = LinearLayoutManager(context)
        binding.historyList.layoutManager = layoutManager
        val adapter = HistoryAdapter()
        binding.historyList.adapter = adapter

        model.getHistory().observe(
            viewLifecycleOwner,
            Observer<List<Action>> { data: List<Action> ->
                adapter.setData(
                    data
                )
            })
    }

    companion object {
        fun newInstance(): HistoryFragment = HistoryFragment()
    }
}
